protocols/imap/src/main/java/org/apache/james/imap/processor/GetAnnotationProcessor.java

```diff
@@ -0,0 +1,163 @@
+import org.apache.james.mailbox.model.MailboxPath;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+
+public class GetAnnotationProcessor extends AbstractMailboxProcessor<GetAnnotationRequest> implements CapabilityImplementingProcessor {
+    public GetAnnotationProcessor(ImapProcessor next, MailboxManager mailboxManager, StatusResponseFactory factory) {
+        super(GetAnnotationRequest.class, next, mailboxManager, factory);
+    }
+
+    public List<String> getImplementedCapabilities(ImapSession session) {
+        return ImmutableList.of(ImapConstants.SUPPORTS_ANNOTATION);
+    }
+
+    protected void doProcess(GetAnnotationRequest message, ImapSession session, String tag, ImapCommand command,
+            Responder responder) {
+        try {
+            proceed(message, session, tag, command, responder);
+        } catch (MailboxNotFoundException e) {
+            session.getLog().info("The command: {} is failed because not found mailbox {}", command.getName(), message.getMailboxName());
+            no(command, tag, responder, HumanReadableText.FAILURE_NO_SUCH_MAILBOX, ResponseCode.tryCreate());
+        } catch (MailboxException e) {
+            session.getLog().info("The command: {} on mailbox {} is failed", command.getName(), message.getMailboxName());
+            no(command, tag, responder, HumanReadableText.GENERIC_FAILURE_DURING_PROCESSING);
+        }
+    }
+
+    private void proceed(GetAnnotationRequest message, ImapSession session, String tag, ImapCommand command, Responder responder) throws MailboxException {
+        String mailboxName = message.getMailboxName();
+        long maxsize = message.getMaxsize();
+        MailboxPath mailboxPath = buildFullPath(session, mailboxName);
+
+        List<MailboxAnnotation> mailboxAnnotations = getMailboxAnnotations(session, message.getKeys(), message.getDepth(), mailboxPath);
+        Optional<Integer> maximumOversizedSize = getMaximumOversizeSize(mailboxAnnotations, maxsize);
+
+        respond(tag, command, responder, mailboxName, mailboxAnnotations, maxsize, maximumOversizedSize);
+    }
+
+    private void respond(String tag, ImapCommand command, Responder responder, String mailboxName,
+                         List<MailboxAnnotation> mailboxAnnotations, long maxsize, Optional<Integer> maximumOversizedSize) {
+        if (maximumOversizedSize.isPresent()) {
+            responder.respond(new AnnotationResponse(mailboxName, filterItemsBySize(responder, mailboxName, mailboxAnnotations, maxsize)));
+            okComplete(command, tag, ResponseCode.longestMetadataEntry(maximumOversizedSize.get()), responder);
+        } else {
+            responder.respond(new AnnotationResponse(mailboxName, mailboxAnnotations));
+            okComplete(command, tag, responder);
+        }
+    }
+
+    private Optional<Integer> getMaximumOversizeSize(List<MailboxAnnotation> mailboxAnnotations, long maxsize) {
```

Comment:
is there a difference between getMaxSizeOfItemsOversize and getMaxSizeOfItems ? I guess oversized items are the bigger anyway ?